sound/hda/ext/hdac_ext_controller.c

```diff
@@ -280,9 +281,11 @@ int snd_hdac_ext_bus_link_get(struct hdac_bus *bus,
 		 *  HDA spec section 4.3 - Codec Discovery
 		 */
 		udelay(521);
-		bus->codec_mask = snd_hdac_chip_readw(bus, STATESTS);
-		dev_dbg(bus->dev, "codec_mask = 0x%lx\n", bus->codec_mask);
-		snd_hdac_chip_writew(bus, STATESTS, bus->codec_mask);
+		codec_mask = snd_hdac_chip_readw(bus, STATESTS);
+		dev_dbg(bus->dev, "codec_mask = 0x%lx\n", codec_mask);
+		snd_hdac_chip_writew(bus, STATESTS, codec_mask);
+		if (!bus->codec_mask)
+			bus->codec_mask = codec_mask;
```

Comment:
can you clarify in the commit message what the impact of overriding the bus->codec_mask is? the patch looks ok, but it's not clear why it's needed.

Reply:
@plbossart Ack. So basicly my patch to power up/down display, at runtime suspend/resume, depends on codec_mask being correct. If hdac library changes it on its own, our PM runtime flow stops working correctly. We could copy the data to a SOF local variable outside HDAC, but it seems that's not really reasonable -- there doesn't seem to be any reason for hdac_ext_controller.c to modify codec_mask like this.

When sending upstream, this should patch should be first in the series (as later patches depend on it). I can still do this :)

Reply:
ok, I'll approve then and let you send all the i915-related upstream on top of Mark's tree. Please add my reviewed-by tag.

when it's merged, we'll deal with conflicts with Ranjani's changes and send a v2.

Reply:
if there's really no reason for hdac_ext_controller to modify `.codec_mask` maybe that should be fixed instead?